packages/@aws-cdk/aws-applicationautoscaling/test/scalable-target.test.ts

```diff
@@ -59,7 +59,7 @@ describe('scalable target', () => {
 
     // WHEN
     target.scaleOnSchedule('ScaleUp', {
-      schedule: appscaling.Schedule.rate(cdk.Duration.minutes(1)),
+      schedule: cdk.Schedule.rate(cdk.Duration.minutes(1)),
```

Comment:
How is it that this works without you having updated the type of the `schedule` property? Must be that that purely works because of TypeScript structural typing, which is not good enough for our Java customers, for example.

Make `appscaling.Schedule extends cdk.Schedule` and change the property type to `cdk.Schedule`.

Reply:
You are right.

But if `appscaling.Schedule extends cdk.Schedule` then the static methods must return a `cdk.Schedule` (jsii) which is a breaking change because `cdk.Schedule` doesn't extend `appscaling.Schedule`?

Reply:
Oh hot damn.